configureWorkspace/configure_dotnetcore.ts

```diff
@@ -0,0 +1,224 @@
+    }
+
+    let serviceName = path.basename(serviceNameAndRelativePath);
+    let projectDirectory = path.dirname(serviceNameAndRelativePath);
+    let projectFileName = `${serviceName}.csproj`;
+    // We don't want the project folder in $assembly_name$ because the assembly is in /app and WORKDIR has been set to that
+    let assemblyNameNoExtension = serviceName;
+    // example: COPY Core2.0ConsoleAppWindows/Core2.0ConsoleAppWindows.csproj Core2.0ConsoleAppWindows/
+    let copyProjectCommands = `COPY ["${serviceNameAndRelativePath}.csproj", "${projectDirectory}/"]`
+    let exposeStatements = `EXPOSE ${port}`;
+
+    // Parse version from TargetFramework
+    // Example: netcoreapp1.0
+    const defaultNetCoreVersion = '2.1';
+    let [netCoreAppVersion] = extractRegExGroups(version, /^netcoreapp([0-9.]+)$/, [defaultNetCoreVersion]);
+
+    // semver requires a patch in the version, so add it if only major.minor
+    if (netCoreAppVersion.match(/^[^.]+\.[^.]+$/)) {
+        netCoreAppVersion += '.0';
+    }
+
+    let baseImageFormat: string;
+    let sdkImageNameFormat: string;
+
+    if (platform === 'ASP.NET Core') {
+        // See https://github.com/aspnet/Announcements/issues/298 - 2.1 and newer use microsoft/dotnet repo
+        if (semver.lt(netCoreAppVersion, '2.1.0')) {
+            baseImageFormat = AspNetCoreRuntimeImageFormat;
+            sdkImageNameFormat = AspNetCoreSdkImageFormat;
+        } else {
+            baseImageFormat = DotNetCoreAspNetRuntimeImageFormat;
+            sdkImageNameFormat = DotNetCoreSdkImageFormat;
+        }
+    } else {
+        assert.equal(platform, '.NET Core Console');
+        baseImageFormat = DotNetCoreRuntimeImageFormat;
+        sdkImageNameFormat = DotNetCoreSdkImageFormat;
+    }
+
+    // When targeting Linux container or the dotnet core version is less than 2.0, use MA tag.
+    // Otherwise, use specific nanoserver tags depending on Windows build.
+    let tagForWindowsVersion: string;
+    if (os === 'Linux' || semver.lt(netCoreAppVersion, '2.0.0')) {
+        tagForWindowsVersion = '';
+    } else {
+        tagForWindowsVersion = GetWindowsImageTag();
+    }
+
+    let baseImageName = formatVersion(baseImageFormat, netCoreAppVersion, tagForWindowsVersion);
+    let sdkImageName = formatVersion(sdkImageNameFormat, netCoreAppVersion, tagForWindowsVersion);
+
+    let template: string;
+    switch (platform) {
+        case ".NET Core Console":
+            template = os === "Linux" ? dotNetCoreConsoleLinuxTemplate : dotNetCoreConsoleWindowsTemplate;
+            break;
+        case "ASP.NET Core":
+            template = os === "Linux" ? aspNetCoreLinuxTemplate : aspNetCoreWindowsTemplate;
+            break;
+        default:
+            throw new Error(`Unexpected platform "${platform}"`);
+    }
+
+    let contents = template.replace('$base_image_name$', baseImageName)
+        .replace(/\$expose_statements\$/g, exposeStatements)
```

Comment:
It it not clear to me whether #391 would be handled. There is no conditional logic in inserting the `$expose_statements$`. It is present b default on both aspnetCore templates,  and absent by default on both dotnetCoreTemplates

Reply:
It's always present in the template now, but exposeStatements will be empty string if no port.  As you see, I handled this more thoroughly for other platforms in https://github.com/Microsoft/vscode-docker/pull/490 but don't plan to merge until after shipping.

Reply:
(by now, I mean with the PR that came after this)

Reply:
Agreed. Thanks for the fix there! I was just pointing out a bug we triaged.